src/components/molecules/FolderNavigatorItem.tsx

```diff
@@ -224,6 +239,81 @@ const FolderNavigatorItem = ({
     }
   }
 
+  const handleDragOverNavigatorItem = useCallback((e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    e.dataTransfer.dropEffect = 'move'
+    setDraggedOver(true)
+  }, [])
+
+  const handleDragLeave = (e: React.DragEvent) => {
+    e.preventDefault()
+    e.stopPropagation()
+    console.log('Drag is over!')
+    setDraggedOver(false)
+  }
+
+  const handleDragStartNavigatorItem = (event: React.DragEvent) => {
+    setTransferableTextData(event, folderPathnameFormat, folderPathname)
+  }
+
+  const handleDropFolderToNavigatorItem = (
+    sourceItemPathname: string,
+    destinationFolderPathname: string
+  ) => {
+    const sourceFolderName = sourceItemPathname.substring(
+      sourceItemPathname.lastIndexOf('/')
+    )
+    const newFolderPathname = `${destinationFolderPathname}${sourceFolderName}`
+    renameFolder(
+      storageId,
+      sourceItemPathname == '' ? '/' : sourceItemPathname,
+      newFolderPathname
+    )
+      .then(() => {
+        // refresh works for file system (FSNote) database
+        push(`/app/storages/${storageId}/notes${newFolderPathname}`)
+        openSideNavFolderItemRecursively(storageId, newFolderPathname)
+      })
+      .catch((err) => {
+        pushMessage({
+          title: 'Folder Structure Update Failed',
+          description: `Updating folder location failed. Reason: ${
+            err != null && err.message != null ? err.message : 'Unknown'
+          }`,
+        })
+      })
+  }
+  const handleDropNoteToNavigatorItem = (
+    noteId: string,
+    destinationFolderPathname: string
+  ) => {
+    // move folder pathname of the note to new location (target destination)
+    updateNote(storageId, noteId, {
+      folderPathname: destinationFolderPathname,
+    })
+  }
+
+  const handleDropNavigatorItem = (event: React.DragEvent) => {
```

Comment:
This code repeats on 3 places, but not sure how to refactor - should I send `updateNote` and `renameFolder` functions to a utility function and then it can handle all details for all three usages (on note/folder/workspace drop)?